File: app/src/main/java/com/handydev/financisto/app/FinancistoApp.java
package com.handydev.financisto.app;

import android.content.Context;
import android.content.res.Configuration;

import androidx.multidex.MultiDexApplication;

import com.handydev.financisto.export.drive.GoogleDriveClient;
import com.handydev.financisto.utils.MyPreferences;

import org.androidannotations.annotations.AfterInject;
import org.androidannotations.annotations.Bean;
import org.androidannotations.annotations.EApplication;
import org.greenrobot.eventbus.EventBus;

@EApplication
public class FinancistoApp extends MultiDexApplication {

    public EventBus bus;

    @Bean
    public GoogleDriveClient driveClient;

    @AfterInject
    public void init() {
        bus = EventBus.getDefault();
        //bus.register(driveClient);
    }

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(MyPreferences.switchLocale(base));
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
        MyPreferences.switchLocale(this);
    }
}

File: app/src/main/java/com/handydev/main/FinancistoApp.kt
package com.handydev.main

import com.handydev.financier.MyEventBusIndex
import com.handydev.financier.app.FinancistoApp
import org.greenrobot.eventbus.EventBus

class FinancistoApp: FinancistoApp() {
    override fun onCreate() {
        super.onCreate()
        val eventBus: EventBus = EventBus.builder().addIndex(MyEventBusIndex()).build()
        EventBus.builder().addIndex(MyEventBusIndex()).installDefaultEventBus()
    }
}